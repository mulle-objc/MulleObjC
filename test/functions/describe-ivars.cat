#!/bin/sh

sed -e 's/0x\([0-9a-fA-F][0-9a-fA-F]*\)/<pointer>/g' "$@"