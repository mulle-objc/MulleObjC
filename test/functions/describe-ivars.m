#import <MulleObjC/MulleObjC.h>



@interface Bar  : NSObject

@property( assign) NSUInteger  count;
@property( assign) float       borderWidth;
@property( retain) Bar         *object;
@property( retain) Bar         *nilObject;
@property( assign) char        *string;
@property( assign) char        *nullString;

@end 


@implementation Bar 

- (char *) UTF8String
{
   return( "VfL Bochum 1848");
}

@end 


static void   test_bar( void)
{
   Bar   *x = [Bar object];
   Bar   *y = [Bar object];
   char  *s;

   [x setObject:y];
   [x setCount:1848];
   [x setBorderWidth:18.48];
   [x setString:"Vfl Bochum 1848"];

   mulle_buffer_do( buffer)
   {
      MulleObjCDescribeIvars( buffer, x);
      s = mulle_buffer_get_string( buffer);
      printf( "%s\n", s);
   }
}


int   main( int argc, char *argv[])
{
   test_bar();

   return( 0);
}

