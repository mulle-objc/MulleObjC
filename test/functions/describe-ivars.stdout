{
   _borderWidth = 18.48,
   _count = 1848,
   _nilObject = nil,
   _nullString = NULL,
   _object = <Bar <pointer>>,
   _string = "Vfl Bochum 1848"
}
